README.md

```diff
@@ -107,46 +112,72 @@ The general idea of portals is summed up above: inset pre-rendering, activation,
 
-#### Channels that match navigation
+If we allowed access to unpartitioned storage, then side channels available pre-activation (e.g., server-side timing correlation) could potentially be used to join two separate user identifiers, one from the host site and one from the portaled site's unpartitioned storage.
 
-* The URL of the activated portal and the referring URL are available to normal navigations to the same extent they're available to a portal. Solutions to link decoration will apply to both.
+The below subsections explore the implementation of these restrictions in more detail.
 
+#### Storage access blocking
 
-TODO:
+Portaled pages that are cross-origin to their host will have no access to storage, similar to how an opaque-origin `<iframe>` behaves. (See this [discussion on the spec mechanism](https://github.com/whatwg/storage/issues/18#issuecomment-615336554).)
+
+We could attempt to address the threat by providing partitioned or ephemeral storage access, but then it is unclear how to transition to _unpartitioned_ storage upon activation. It would likely require some kind of web-developer-written merging logic. Completely blocking storage access is thus deemed simpler; portaled pages should not be doing anything which requires persistent storage before activation.
+
+This means that most existing content will appear "broken" when portaled by a cross-origin host. This necessitates an explicit opt-in to allow cross-origin content to be portaled, [discussed below](#opt-in). Such content might optionally "upgrade" itself to a credentialed view upon activation, as shown in the [example](#example) above.
+
+For a more concrete example, consider `https://aggregator.example/` which wants to prerender this GitHub repository using a `<portal>`. To make this work, GitHub would need to add the opt-in to allow the page to be portaled. Additionally, GitHub should add code to adapt their UI to show the logged-in view upon activation, by removing the "Join GitHub today" banner, and retrieving the user's credentials from storage and using them to replace the signed-out header with the signed-in header. Without such adapter code, activating the portal would show the user a logged-out view of GitHub in the top-level tab that the portal has been activated into. This would be a bad and confusing user experience, since the user is logged in to GitHub in all of their other top-level tabs.
+
+#### Communications channels that are blocked
+
+- [`postMessage()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) isn't allowed between a cross-origin portal and its container.
+- Blocked by preventing fetches within portals from using credentials until they're activated:
```

Comment:
Drive-by: I was reading the explainer today and found this section confusing so this review seems like an ideal time to ask questions and make clarifications.

> Blocked by preventing fetches within portals from using credentials until they're activated

This sentence is awkward. Does "Fetches within an unactivated portal are performed without credentials" describe what this is saying?

Reply:
Also, are the two sub-bullets below examples of potential communication channels if a portal can perform credentialed requests? If so, perhaps:

"Fetches within an unactivated portal are performed without credentials. Credentialed fetches can be used for cross-site recognition, for example:"

Reply:
Also, also, presumably this is all moot for same-origin fetches - do we still fetch uncredentialed? What about the other restrictions below?

Reply:
I'll clarify this in the manner you suggest; I agree these are good changes.